server/src/main/java/com/jme3/math/Quaternion.java

```diff
@@ -923,6 +924,74 @@ public void nlerp(Quaternion q2, float blend) {
 		normalizeLocal();
 	}
 
+	/**
+	 * Sets the values of this quaternion to the unit average of the given
+	 * Quaternions and return itself
+	 *
+	 * @param qn list of all the Quaternions to average
+	 * @return a new Quaternion resulting from the average of the given
+	 * Quaternions
+	 */
+	public Quaternion fromAveragedQuaternions(AbstractList<Quaternion> qn) {
+		if (qn.size() == 0) {
+			throw new IllegalArgumentException("qn's length can't be 0'");
+		}
+
+		float sumX = 0f;
+		float sumY = 0f;
+		float sumZ = 0f;
+		float sumW = 0f;
+		for (Quaternion q : qn) {
+			sumX += q.x;
+			sumY += q.y;
+			sumZ += q.z;
+			sumW += q.w;
+		}
+
+		x = sumX / qn.size();
+		y = sumY / qn.size();
+		z = sumZ / qn.size();
+		w = sumW / qn.size();
+
+		return this;
+	}
+
+	/**
+	 * Sets the values of this quaternion to the unit average of the given
+	 * Quaternions and return itself
+	 *
+	 * @param qn list of all the Quaternions to average
+	 * @param tn 0-1, averaging weight for Quaternions (sum = 1)
+	 * @return a new Quaternion resulting from the weighted average of the given
+	 * Quaternions
+	 */
+	public Quaternion fromAveragedQuaternions(AbstractList<Quaternion> qn, AbstractList<Float> tn) {
+		if (qn.size() == 0) {
+			throw new IllegalArgumentException("qn's length can't be 0'");
```

Comment:
Same as my last comment
```suggestion
			throw new IllegalArgumentException("qn's length can't be 0");
```